applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/ApplicationLinksAppService.cs

```diff
@@ -45,27 +43,35 @@ public async Task<List<ApplicationLinksInfoDto>> GetListByApplicationAsync(Guid
         var applications = await applicationsTask;
         var applicationForms = await applicationFormsTask;
 
-        var combinedQuery = from applicationLink in applicationLinks
-                            join application in applications on applicationLink.ApplicationId equals application.Id into appGroup1
-                            from application in appGroup1.DefaultIfEmpty()
-                            join application2 in applications on applicationLink.LinkedApplicationId equals application2.Id into appGroup2
-                            from application2 in appGroup2.DefaultIfEmpty()
-                            join appForm in applicationForms on application.ApplicationFormId equals appForm.Id into appFormGroup
-                            from appForm in appFormGroup.DefaultIfEmpty()
-                            join appForm2 in applicationForms on application2.ApplicationFormId equals appForm2.Id into appFormGroup2
-                            from appForm2 in appFormGroup2.DefaultIfEmpty()
-                            where (applicationLink.ApplicationId == applicationId && application != null)
-                               || (applicationLink.LinkedApplicationId == applicationId && application2 != null)
-                            select new ApplicationLinksInfoDto
-                            {
-                                Id = applicationLink.Id,
-                                ApplicationId = application != null ? application.Id : application2.Id,
-                                ApplicationStatus = application != null ? application.ApplicationStatus.InternalStatus : application2.ApplicationStatus.InternalStatus,
-                                ReferenceNumber = application != null ? application.ReferenceNo : application2.ReferenceNo,
-                                Category = application != null ? appForm.Category : appForm2.Category,
-                                ProjectName = application != null ? application.ProjectName : application2.ProjectName
-                            };
+        var query1 = from applicationLink in applicationLinks
```

Comment:
We can further improve this by actually implementing the query through the repository - we have found that this joining through the IQueryables can be very slow - so essentially create a method on the ApplicationLinks Repository and work directly with the dbset at that level